Add docs on solving exercises and feedback (#181)
exercism/docs: using/solving-exercises/README.md

# Solving Exercises

The primary way in which Exercism helps you learn is by solving exercises.

There are two ways to solve exercises on Exercism:

> 1. [Online via the website's in-browser editor](./online.md)

Reviewer: ```suggestion
1. [Online via the website's in-browser editor](./online)
```

2. [On your local machine via the CLI](./locally)

## Which option should I choose?

There are different trade-offs between using the online editor or CLI:

|            | Online editor                          | CLI                             |
| ---------- | -------------------------------------- | ------------------------------- |
| Installing | No                                     | Yes (e.g. SDK)                  |

> | Editing    | Browser                                | Your preferred IDE              |

Reviewer: ```suggestion
| Editing    | Browser                                | Your preferred text editor      |
```
Better to be more general here? Not everyone uses an IDE, but an IDE contains a text editor.

Author: Or maybe just "editor"?

| Testing    | Run tests on website                   | Run tests locally               |

> | Debugging  | Limited support (only for some tracks) | Full support (depending on IDE) |

Author: I'm hoping this helps people choose between the two options.

> | Debugging  | Limited support (only for some tracks) | Full support (depending on IDE) |

Reviewer: I'd also consider removing the reference to "IDE" here too, since people might prefer to debug without using an IDE.

Author: Maybe:

```suggestion
| Debugging  | Limited support (only for some tracks) | Full support (depending on your editor) |
```

Reviewer: Mmm. I think I prefer something like:
> Full support (depending on your machine)

> Full support (depending on your environment)

But I'm fine with whatever you want here.

Author: > Full support (depending on your environment)

This is the best option I think. I'll update
